.clangd

```diff
@@ -1,2 +1,39 @@
 CompileFlags:
-  CompilationDatabase: build/       # Search build/ directory for compile_commands.json
+  # Search build/ directory for compile_commands.json
+  CompilationDatabase: build
+---
+# Additional useful checks that are too noisy to be included in .clang-tidy
+If:
+  PathMatch: src/nvim/api/.*
```

Comment:
No, this will make it so that even useful warnings (some of which I rely on) won't show for the regular files since this file takes priority over users own configuration.

This PR is close to being finished, I suggest not making any more changes unless necessary so that we can get this merged. We can iterate on this further in a follow-up PR.

Reply:
just so we're on the same page, this `If` only restricts the following fragments, this means: 
- src/nvim/api: all configured clang-tidy checks +  these defined here (misc, perf, etc)
- everything else: only use configured clang-tidy checks

<details>
<summary>Demo: notice how the "readability-identifier-naming" is only firing for the file in "src/nvim/api"
</summary>

![image](https://user-images.githubusercontent.com/59826753/195060549-cdc633ed-b344-4ff8-b086-69e41053154e.png)


</details>

given the above,  you still want me to revert this and make these checks global?

Reply:
Yes, because there are useful warnings in `.clang-tidy` that should be in `.clangd` instead, which are now ignored. Also, I don't understand why we're separating `src/nvim/api` with the rest of the files, shouldn't they have the same configuration?

Reply:
Do we actually need a `.clang-tidy`? This will be much easier if we just add only necessary warnings to `.clangd` and then add stuff to `.clang-tidy` in a future PR.

Reply:
> Yes, because there are useful warnings in `.clang-tidy` that should be in `.clangd` instead, which are now ignored.

@dundargoc, this should not be the case, since `.clangd` is not overriding `.clang-tidy` here, just as I explained earlier.

can you please provide an example where that is happening? I've tested on both v14 and v15 and I can't replicate this.

![image](https://user-images.githubusercontent.com/59826753/195108178-1ebd7603-89c2-4c97-a6be-3063cf3bb64b.png)

> Do we actually need a `.clang-tidy`?

yeah we do since clangd doesn't support all the options AFAIK, see https://github.com/clangd/clangd/issues?q=tidy+is%3Aopen

Reply:
`src/nvim/memline.c` is a good example: there are a bunch of unnecessary casts there and this PR removes those warnings.

Reply:
oh, okay I think I can see that you mean `readability-non-const-parameter` is useful, where you'd still want it on for all the files.
either way, I'll remove the guard since we can always put it back if it's too noisy.

Reply:
> yeah we do since clangd doesn't support all the options AFAIK, see https://github.com/clangd/clangd/issues?q=tidy+is%3Aopen

@kylo252 what am I supposed to look at here?